lua/apisix/balancer.lua

```diff
@@ -230,6 +231,13 @@ local function pick_server(route, ctx)
         key = up_conf.type .. "#route_" .. route.value.id
     end
 
+    if up_conf.service_name then
+        if not discovery then
+            return nil, nil, "discovery is uninitialized"
+        end
+        up_conf.nodes = discovery.nodes(up_conf.service_name)
```

Comment:
We had better not change the original data. 

Here is an example, please take a look:
https://github.com/apache/incubator-apisix/blob/master/lua/apisix.lua#L205

One more thing, are the node addresses in IP format?
If the node's address is a domain name, then we need to resolve the IP address first in the rewrite or access phase.

The domains cannot be used directly in the Nginx balancer phase.

Reply:
where could change the nodes?

 the node addresses is IP format.

Reply:
we can store it to `route.dns_value`

you can take a look at:
https://github.com/apache/incubator-apisix/blob/master/lua/apisix/balancer.lua#L202